lib/active_merchant/billing/gateways/world_net_v2/token.rb

```diff
@@ -0,0 +1,75 @@
+require 'active_merchant/billing/base'
+require 'active_merchant/posts_data'
+require 'json'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    module WorldNetV2 #:nodoc:
+      class Token
+        include PostsData
+
+        TerminalMissingError = Class.new(StandardError)
+
+        @lock = Mutex.new
+        class_attribute :merchant_api_key, instance_writer: false
+        class_attribute :token_data, instance_writer: false
+        class_attribute :token_expires_at, instance_writer: false
```

Comment:
I think this might imply that only one API Key for worldnet gateway is being used at a time.

If we had multiple merchant accounts, I believe we would have multiple API keys.  

Might need to check with Mark on that.

Reply:
Yes, that's a good point. If one application process could be handling processing for more than one merchant, then I can change how the token works. Right now it's a singleton.